### PR TITLE
initial commit for a pcs module/state

### LINKED ISSUE #31858: A module and a state for managing pcs (pacemaker configuration system)
I created a module and a state for managing pacemaker/corosync clusters via pcs. 
In the guidelines for PR's it is mentioned that an issue should be created first for discussions regarding the PR.
Here it is :)
PR follows...

### PR DESCRIPTION
**Pull request by ghost**
This PR adds a new module and state for managing pacemaker/corosync-clusters via pcs.

It's in an early stage and it is the 1st python project I do and my 1st contribution to salt.

Currently it's possible to 
- setup authentication for cluster nodes
- setup/start/enable pacemaker/corosync on cluster nodes
- setup a stonith fencing resource in the cluster

To see a working formula for managing pcs via salt and this module/statemodule see https://github.com/hoonetorg/salt-pcs-formula.

Fixes #31858.
### Tests written?
- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by cachedout** [comment]
Hi @hoonetorg 

Thanks for this contribution! There are a number of style issues which need to be corrected just to ensure that the Python you've written meets our guidelines. Could you take a look at this report and make some updates to your PR? 
https://jenkins.saltstack.com/job/salt-pr-lint-n/14234/violations

Once you've made those fixes and pushed them up, we'll go through this and see if we can it merged. Thanks.

**Comment by hoonetorg** [comment]
I will fix that
is there a good tool for checking lint directly on linux (in vim f.e.)

**Comment by rallytime** [comment]
@hoonetorg I am not sure of an example to use in vim directly, but you can run pylint manually on any files you're changing before submitting your pull request. Check out the [Linting](https://docs.saltstack.com/en/latest/topics/development/conventions/style.html#linting) section of the docs.

**Comment by rallytime** [comment]
Go Go Jenkins!

**Comment by cachedout** [comment]
@hoonetorg There are a number of cases here where you're using lists as default kwargs. We can't have mutable types as keyword arguments. Please change these to `None` and then modify them as necessary in the function's implementation. Thanks.

**Comment by hoonetorg** [comment]
@cachedout: is the change with kwargs also ok?

I only want to get the lint complaints done. There is probably more to be done, which we can discuss later.

**Comment by hoonetorg** [comment]
O.K. lint passes. Is that O.K.?
What else needs to be discussed?

File: salt/modules/pcs.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+def __virtual__():
+    '''
+    Only load if pcs is installed
+    '''
+    if salt.utils.which('pcs'):
+        return 'pcs'
+    return False
+
+
+def auth(nodes, pcsuser='hacluster', pcspasswd='hacluster', **kwargs):
```
Comment (cachedout) [inline salt/modules/pcs.py]: If we're going to document the use of `extra_args` let's use that explicitely versus using `kwargs`

**Comment by cachedout** [comment]
@hoonetorg Replied with one more comment in-line. Please have a look.

**Comment by cachedout** [comment]
@hoonetorg Bump. Did you see the above comments?

**Comment by hoonetorg** [comment]
Sorry, many real-life todo's this week. Back on the track tomorrow morning

**Comment by cachedout** [comment]
@hoonetorg Many thanks!

**Comment by hoonetorg** [comment]
@cachedout cachedout: OK?
What now?
Do I need to write tests?

File: salt/modules/pcs.py [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+        cmd += ['-u', pcsuser]
+
+    if pcspasswd:
+        cmd += ['-p', pcspasswd]
+
+    if isinstance(extra_args, (list, tuple)):
+        cmd += extra_args
+    cmd += nodes
+
+    return __salt__['cmd.run_all'](cmd, output_loglevel='trace', python_shell=False)
```
Comment (hoonetorg) [inline salt/modules/pcs.py]: @cachedout @rallytime: 
Are there rules for return values of module functions?
Would it be better to make this here an internal function (_auth) for use with the state module pcs.auth and create a new function with return values True or False (or whatever are the rules for return values of module functions)?

Comment (cachedout) [inline salt/modules/pcs.py]: We prefer that they are not `None` but you should be OK in this case.

**Pull request merged**